src/mcMd/species/LinearSG.cpp

```diff
@@ -0,0 +1,399 @@
+/*
+* Simpatico - Simulation Package for Polymeric and Molecular Liquids
+*
+* Copyright 2010 - 2014, The Regents of the University of Minnesota
+* Distributed under the terms of the GNU General Public License.
+*/
+
+#include <mcMd/simulation/System.h>
+#include <mcMd/simulation/Simulation.h>
+#include <mcMd/potentials/bond/BondPotential.h>
+#include <mcMd/neighbor/CellList.h>
+#include <mcMd/chemistry/Molecule.h>
+#include "LinearSG.h"
+#ifdef UTIL_MPI
+#include <mcMd/simulation/McMd_mpi.h>
+#endif
+
+namespace McMd
+{
+
+   using namespace Util;
+
+   /*
+   * Constructor.
+   */
+   LinearSG::LinearSG()
+    : Linear(),
+      SpeciesMutator(),
+      beadTypeIds1_(),
+      beadTypeIds2_()
+   {
+      setMutatorPtr(this);
+   }
+
+   /*
+   * Destructor.
+   */
+   LinearSG::~LinearSG()
+   {}
+
+   /*
+   * Call general Species::readParameters() .
+   * Doesn't seem to do anything
+   
+   void LinearSG::readParameters(std::istream& in)
+   {
+      Species::readParameters(in);
+   }
+   */
+
+   /*
+   * Read atom structure and two sets of atom type ids.
+   */
+   void LinearSG::readSpeciesParam(std::istream& in)
+   {
+      read<int>(in,"nAtom", nAtom_);
+      nBond_ = nAtom_ - 1;
+      #ifdef INTER_ANGLE
+      nAngle_ = nAtom_ - 2;
+      #endif
+      #ifdef INTER_DIHEDRAL
+      if (nAtom_ > 3)
```

Comment:
Need to read hasDihedrals_ and (if hasDihedrals) dihedralType_, as in HomopolymerS class.